File: mod5/src/create_bucket.py
# 依存関係をインポート
import boto3
import uuid

# 設定
region = 'ap-northeast-1'
bucket_name = f'sample-bucket-{uuid.uuid4()}'
print(f'作成するバケット名: {bucket_name}')

# S3 クライアントの作成
s3_client = boto3.client('s3', region_name=region)

# バケットの作成
s3_client.create_bucket(
    Bucket=bucket_name,
    CreateBucketConfiguration={
        'LocationConstraint': region
    }
)

File: mod5/src/create_bucket.java
package com.mycompany.app;

// 依存関係をインポート
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;

import java.util.UUID;

public class App {
    public static void main(String[] args) {
        // 設定
        String region = "ap-northeast-1";
        String bucket_name = "sample-bucket-" + UUID.randomUUID();
        System.out.println("作成するバケット名:" + bucket_name);

        //S3 クライアントの作成
        AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withRegion(region).build();

        // バケットの作成
        s3Client.createBucket(bucket_name);

        // 明示的にクライアントをシャットダウン
        s3Client.shutdown();
    }
}
